rajawali/src/main/java/c/org/rajawali3d/core/ACoreComponent.java

```diff
@@ -0,0 +1,160 @@
+package c.org.rajawali3d.core;
+
+import c.org.rajawali3d.annotations.RenderThread;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.FloatRange;
+
+import net.jcip.annotations.GuardedBy;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Base class for all core components; enables common render thread tasking
+ *
+ * @author Randy Picolet
+ */
+
+abstract class ACoreComponent implements CoreComponent {
```

Comment:
I don't particularly care what naming convention is used, but what should we use when there is both an interface and an abstract shared implementation of that interface, as there is here?